src-tauri/build.rs

```diff
@@ -1,97 +1,69 @@
 use eyre::{bail, Result};
 use flate2::read::GzDecoder;
 use reqwest::get;
-use std::env;
-use std::env::consts::{ARCH, OS};
+use shared_utils::determine_bin_data;
 use std::fs;
 use std::fs::File;
 use std::io::copy;
 use std::io::BufReader;
 use std::io::Read;
+use std::path::PathBuf;
 use tar::Archive;
 
 fn main() {
     tauri::async_runtime::block_on(setup_binary()).unwrap();
     tauri_build::build()
 }
 
-pub async fn setup_binary() -> Result<()> {
-    let target = determine_target();
+async fn setup_binary() -> Result<()> {
+    let (os, arch, target) = determine_bin_data();
     let binary_name = "meroctl";
-    let cache_dir = std::env::temp_dir().join("meroctl");
+    let cache_dir = std::env::temp_dir().join(binary_name);
     std::fs::create_dir_all(&cache_dir).expect("Failed to create cache directory");
 
     let url = format!(
         "https://github.com/calimero-network/core/releases/latest/download/{}.tar.gz",
         target
     );
-    let binary_path = cache_dir.join(format!("{}.tar.gz", binary_name));
-
-    let current_dir = env::current_dir().expect("Failed to get current directory");
-    let bin_dir = current_dir.join("bin").join(OS).join(ARCH);
+    let cache_bin_path = cache_dir.join(format!("{}.tar.gz", binary_name));
+    let bin_dir = std::env::current_dir()?.join("bin").join(os).join(arch);
     let resource_path = bin_dir.join(binary_name);
 
     if resource_path.exists() {
         return Ok(());
     }
 
-    let response = get(&url).await.expect("Failed to download binary");
-    let mut out = File::create(&binary_path).expect("Failed to create file");
+    download_and_extract(&url, &cache_bin_path, &bin_dir).await?;
+
+    Ok(())
+}
+
+async fn download_and_extract(
+    url: &str,
+    cache_bin_path: &PathBuf,
+    bin_dir: &PathBuf,
```

Comment:
```suggestion
    cache_bin_path: &Path,
    bin_dir: &Path,
```

`&Path` is to `PathBuf` what `&str` is to `String`

Reply:
aaaaa nice! ok thank you!